### PR TITLE
feat: support secrets in render task definition

### PR DESCRIPTION
**Pull request by kg-aws**
*Issue #, if available:*
#20 
*Description of changes:*
NOTE: This PR is the follow up / rebased version of the PR: https://github.com/aws-actions/amazon-ecs-render-task-definition/pull/152

Add ability to add secrets to task definition. There is already a pull request here https://github.com/aws-actions/amazon-ecs-render-task-definition/pull/143 which tries to do this, but it isn't backward compatible. This pull request is. It also doesn't add tests which this PR does.

Thank you @yehudacohen and @hectorgrecco for the initial PR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ async function run() {
     const taskDefinitionFile = core.getInput('task-definition', { required: false });
     const containerName = core.getInput('container-name', { required: true });
     const imageURI = core.getInput('image', { required: true });
-    const environmentVariables = core.getInput('environment-variables', { required: false });
-    const envFiles = core.getInput('env-files', { required: false });
+    const environmentVariables = core.getInput('environment-variables', { required: false });    
+    const envFiles = core.getInput('env-files', { required: false });
```
Comment (amazreech) [inline index.js:18]: nit: Looks like unintentional extra lines are added at the end of these lines?

File: index.js [hunk 2]
```diff
@@ -27,6 +27,8 @@ async function run() {
     const taskDefinitionFamily = core.getInput('task-definition-family', { required: false }) || undefined;
     const taskDefinitionRevision = Number(core.getInput('task-definition-revision', { required: false })) || null;
```
Comment (amazreech) [inline index.js:29]: nit: extra line

File: index.js [hunk 3]
```diff
@@ -135,6 +137,45 @@ async function run() {
...
+          if (trimmedLine.length === 0) { return; }
+          // Split on =
+          const separatorIdx = trimmedLine.indexOf("=");
+          // If there's nowhere to split
+          if (separatorIdx === -1) {
+              throw new Error(
+                `Cannot parse the secret '${trimmedLine}'. Secret pairs must be of the form NAME=valueFrom, 
+                where valueFrom is an arn from parameter store or secrets manager. See AWS documentation for more information: 
+                https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html.`);
+          }
```
Comment (amazreech) [inline index.js:161]: These lines are identical with the lines 113 to 122 above. We could extract and create a common method that does this functionality of trimming and splitting?

**Review by amazreech — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (kg-aws) [reply index.js]: Fixed the extra line. Resolving the comment.

File: index.js (hunk 2 above)
Comment (kg-aws) [reply index.js]: Fixed the extra line. Resolving the comment.

**Review by amazreech — APPROVED** [review]
Thanks Kishan, LGTM!

**Comment by kurarrr** [comment]
It would be appreciated if you could release this feature. Thanks in advance!

**Comment by Ma-hawaj** [comment]
I see that this feature is released but I'm having issues with it.
I'm using the following:
```yaml
      - name: Fill in the new image ID in the Amazon ECS task definition
        id: task-def-dev
        uses: aws-actions/amazon-ecs-render-task-definition@v1.6.0
        with:
          task-definition-family: dev-chatbot-client-family
          container-name: dev-chatbot-client-container
          secrets: |
            BOT_API_KEY=arn:aws:secretsmanager:***
            BOT_URL=arn:aws:secretsmanager:***
            SUPPORT_CALLBACK_URL=arn:aws:secretsmanager:***
            TABLE_NAME=arn:aws:secretsmanager:***
            COGNITO_CLIENT_ID=arn:aws:secretsmanager:***
            AWS_REGION=arn:aws:secretsmanager:***
          image: '***'

      - name: Show task defenetion
        run: cat ${{ steps.task-def-dev.outputs.task-definition }}
```

but I'm getting the task definition without any secrets could any one help with that ?

**Pull request merged**